packages/piro/src/Piro_TempusSolver_Def.hpp

```diff
@@ -386,198 +385,109 @@ void Piro::TempusSolver<Scalar>::evalModelImpl(
-    if (Teuchos::nonnull(p_in2)) {
-      state_ic.set_p(l+1, p_in2);
-    }
-  }
-
-  //*out << "\nstate_ic:\n" << Teuchos::describe(state_ic, solnVerbLevel);
-
-  //JF  may need a version of the following for multipoint, i.e. num_p>1, l+1, if we want sensitivities
-  RCP<Thyra::MultiVectorBase<Scalar> > dgxdp_out;
-  Thyra::ModelEvaluatorBase::Derivative<Scalar> dgdp_deriv_out;
-  if (num_p > 0) {
-    const Thyra::ModelEvaluatorBase::DerivativeSupport dgxdp_support =
-      outArgs.supports(Thyra::ModelEvaluatorBase::OUT_ARG_DgDp, num_g, l);
-    if (dgxdp_support.supports(Thyra::ModelEvaluatorBase::DERIV_MV_JACOBIAN_FORM)) {
-      const Thyra::ModelEvaluatorBase::Derivative<Scalar> dgxdp_deriv =
-        outArgs.get_DgDp(num_g, l);
-      dgxdp_out = dgxdp_deriv.getMultiVector();
-    }
 
-    if (num_g > 0) {
-      const Thyra::ModelEvaluatorBase::DerivativeSupport dgdp_support =
-        outArgs.supports(Thyra::ModelEvaluatorBase::OUT_ARG_DgDp, j, l);
-      if (!dgdp_support.none()) {
-        dgdp_deriv_out = outArgs.get_DgDp(j, l);
-      }
-    }
-  }
-
-  bool requestedSensitivities = true;
-  if (computeSensitivities_ == true)
-    requestedSensitivities = Teuchos::nonnull(dgxdp_out) || !dgdp_deriv_out.isEmpty();
-  else
-    requestedSensitivities = false;
+  //*out_ << "\nstate_ic:\n" << Teuchos::describe(state_ic, solnVerbLevel_);
 
+  //
+  *out_ << "\nE) Solve the forward problem ...\n";
+  //
   RCP<const Thyra::VectorBase<Scalar> > finalSolution;
   RCP<const Tempus::SolutionState<Scalar> > solutionState;
   RCP<const Tempus::SolutionHistory<Scalar> > solutionHistory;
-  if (!requestedSensitivities)
-  {
-    //
-    *out << "\nE) Solve the forward problem ...\n";
-    //
-    //
-    *out << "T final requested: " << t_final << " \n";
-
-    fwdStateIntegrator->advanceTime(t_final);
-
-    double time = fwdStateIntegrator->getTime();
-
-    *out << "T final actual: " << time << "\n";
-
-    if (abs(time-t_final) > 1.0e-10) {
-      if (abort_on_failure_ == true) {
-        TEUCHOS_TEST_FOR_EXCEPTION(
-          true,
-          Teuchos::Exceptions::InvalidParameter,
-          "\n Error! Piro::TempusSolver: time-integrator did not make it to final time " <<
-          "specified in Input File.  Final time in input file is " << t_final <<
-          ", whereas actual final time is " << time << ".  If you'd like to " <<
-          "suppress this exception, run with 'Abort on Failure' set to 'false' in " << 
-          "Tempus sublist.\n" );
-      }
-      else {
-         *out << "\n WARNING: Piro::TempusSolver did not make it to final time, but "
-              << "solver will not abort since you have specified 'Abort on Failure' = 'false'.\n"; 
-      }
+    
+  *out_ << "T final requested: " << t_final_ << " \n";
+  piroTempusIntegrator_->advanceTime(t_final_);
+  double time = piroTempusIntegrator_->getTime();
+  *out_ << "T final actual: " << time << "\n";
+
+  if (abs(time-t_final_) > 1.0e-10) {

```

Comment:
This looks problem dependent. Should the difference be scaled? A relative error check?

Reply:
Good call, I have changed this to a relative difference.